.github/workflows/coreneuron-ci.yml

```diff
@@ -0,0 +1,162 @@
+          - {use_nmodl: ON, py_version: 3.6.7}
+          - {use_nmodl: ON}
+          - {use_ispc: ON, py_version: 3.6.7}
+          - {gcc_version: 9}
+      fail-fast: false
+
+    steps:
+
+      - name: Install homebrew packages
+        if: startsWith(matrix.os, 'macOS')
+        run: |
+          brew install coreutils bison flex boost openmpi
+        shell: bash
+
+      - name: Install apt packages
+        if: startsWith(matrix.os, 'ubuntu')
+        run: |
+          sudo apt-get install doxygen bison flex libboost-all-dev libopenmpi-dev openmpi-bin python3-dev python3-pip
+        shell: bash
+
+      - name: Install specific apt packages
+        if: startsWith(matrix.os, 'ubuntu') && matrix.config.gcc_version
+        run: |
+          sudo apt-get install g++-${GCC_VERSION}
+        shell: bash
+        env: 
+          GCC_VERSION: ${{ matrix.config.gcc_version }} 
+
+      - name: Set up Python3
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+        env:
+          PYTHON_VERSION:  ${{ matrix.config.py_version || env.DEFAULT_PY_VERSION }}
+
+      - name: Install ISPC
+        if: ${{ matrix.config.use_ispc == 'ON' }}
+        working-directory: ${{runner.workspace}}
+        run: |
+          ispc_version="v1.12.0";
+          if [ "${{ startsWith(matrix.os, 'ubuntu') }}" == "true" ]; then
+            url_os="linux";
+            ispc_version_suffix="b";
+          else
+            url_os="macOS";
+            ispc_version_suffix="";
+          fi;
+          url="https://github.com/ispc/ispc/releases/download/${ispc_version}/ispc-${ispc_version}${ispc_version_suffix}-${url_os}.tar.gz";
+          wget -O ispc.tar.gz $url;
+          mkdir ispc && tar -xvzf ispc.tar.gz -C ispc --strip 1;
+      - name: Install NMODL dependencies
+        if: ${{ matrix.config.use_nmodl == 'ON' ||  matrix.config.use_ispc == 'ON' }}
+        run: |
+          python3 -m pip install --upgrade pip jinja2 pyyaml pytest "sympy<1.6";
+       
+      - uses: actions/checkout@v2
+
+      - name: Install Python3 documentation dependencies
+        if: ${{ startsWith(matrix.os, 'ubuntu') && matrix.config.documentation == 'ON' }}
+        working-directory: ${{runner.workspace}}/CoreNeuron
+        run: |
+          python3 -m pip install --upgrade pip
+          python3 -m pip install --upgrade -r docs/docs_requirements.txt
+        
+      - name: Build and Test
+        id: build-test
+        shell: bash
+        working-directory: ${{runner.workspace}}/CoreNeuron
+        run:  |
+          if [ -n "$GCC_VERSION" ]; then
+              export CXX="g++-${GCC_VERSION}" CC="gcc-${GCC_VERSION}";
+          fi
+          
+          if [[ "${{ startsWith(matrix.os, 'macOS') }}" = "true" ]]; then
+              git submodule update --init -- external/Random123
+              export PATH=${{runner.workspace}}/CoreNeuron/external/Random123:/usr/local/opt/flex/bin:/usr/local/opt/bison/bin:$PATH;
```

Comment:
Curious why random123 is added to PATH?

Reply:
Some strange Random123 is found in the image. Had to put that first

Reply:
Ok..but PATH is for binaries/executables.Random123 is header only library. I wonder how PATH affects that. Do you have build where it was failing?

Reply:
https://github.com/alexsavulescu/CoreNeuron/runs/1464884089?check_suite_focus=true

Reply:
CMake was finding this : `Found Random123: /Users/runner/hostedtoolcache/stack/2.5.1/x64`, and found that to be in the PATH.

Reply:
Quite surprising !